pandas/_libs/tslibs/strptime.pyx

```diff
@@ -532,3 +543,29 @@ cdef tzinfo parse_timezone_directive(str z):
                      (microseconds // 60_000_000))
     total_minutes = -total_minutes if z.startswith("-") else total_minutes
     return pytz.FixedOffset(total_minutes)
+
+cdef int64_t _parse_python_datetime_object(datetime dt, npy_datetimestruct *dts):
+    """
+    Parse a native datetime.datetime object and return a numpy datetime object
+
+    Parameters
+    ----------
+    dt : datetime.datetime instance
+    dts: numpy datetime struct
+
+    Returns
+    -------
+    int64_t
+        the numpy datetime object
+    """
+    dts.year = dt.year
+    dts.month = dt.month
+    dts.day = dt.day
+    dts.hour = dt.hour
+    dts.min = dt.minute
+    dts.sec = dt.second
+    dts.us = dt.microsecond
+    dts.ps = 0  # Not enough precision in datetime objects (https://github.com/python/cpython/issues/59648)
```

Comment:
what if we have a Timestamp object?

Reply:
Fixed using `dts.ps = getattr(dt, "nanosecond", 0) * 1000`

Reply:
This might need a test